src/output.cpp

```diff
@@ -953,6 +951,15 @@ std::string string_input_win(WINDOW *w, std::string input, int max_length, int s
             pos += t.length();
             redraw = true;
         }
+        if( callbacks[ch] ) {
```

Comment:
I put the callback later so we at least to what the window is supposed to do first. Then I moved the escape key to after the callbacks so the function would return before calling the callbacks. It's probably better to put it for yes.